### DISCUSSION TITLE
Lesson 9 Part 7 Error (9582): Member "requestRandomWords" not found or not visible after argument-dependent lookup in address.

### DISCUSSION THREAD

**Discussion by BowtiedHarpyEagle** [Cyfrin/foundry-full-course-cu#1030] [Q&A] [answered]
I'm going through lesson 9 Part 7 Implementing Chainlink VRF at timestamp 10:46, and I'm getting the error:

Error (9582): Member "requestRandomWords" not found or not visible after argument-dependent lookup in address.

Which implies I couldn't import VRFCoordinatorV2Interface properly. 

I have the line 
`import {VRFCoordinatorV2Interface} from "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";`
to import VRFCoordinatorV2Interface.

I did run the command 

`forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit`

without a problem. I see the library is installed. 

I have the line 

`remappings = ['@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/']`

in my foundry.toml file for the remapping. 

What am I missing? 

My full code is at my github repo: 
[my github repo](https://github.com/BowtiedHarpyEagle/foundry-smart-contract-lottery-f23)

**Comment by 0xusmanf** [comment, marked as answer]
[At this line](https://github.com/BowtiedHarpyEagle/foundry-smart-contract-lottery-f23/blob/695e885537b45c28be39118a8635e448c68cae4d/src/Raffle.sol#L41) you have to declare `i_vrfCoordinator` as `VRFCoordinatorV2Interface` instance instead of `address`. 
```solidity
VRFCoordinatorV2Interface private immutable i_vrfCoordinator;
```
And then also assign an `VRFCoordinatorV2Interface` instance in the constructor [at this line](https://github.com/BowtiedHarpyEagle/foundry-smart-contract-lottery-f23/blob/695e885537b45c28be39118a8635e448c68cae4d/src/Raffle.sol#L62).
```solidity
i_vrfCoordinator = VRFCoordinatorV2Interface(vrfCoordinator);
```
Also `REQUEST_CONFIRMATIONS` [here](https://github.com/BowtiedHarpyEagle/foundry-smart-contract-lottery-f23/blob/695e885537b45c28be39118a8635e448c68cae4d/src/Raffle.sol#L36) needs to be type `uint16` and `NUM_WORDS` [here](https://github.com/BowtiedHarpyEagle/foundry-smart-contract-lottery-f23/blob/695e885537b45c28be39118a8635e448c68cae4d/src/Raffle.sol#L37C30-L37C39) needs to be type `uint32`.

**Reply by BowtiedHarpyEagle** [reply]
tx a lot it worked
